File: week_1/day-5-string-hashing/variations_que/most_frequent_character.py
def mostFrequentChar(s):
    freq = {}
    max_char = s[0]
    for ch in s:
        freq[ch] = freq.get(ch, 0) + 1
        if freq[ch] > freq[max_char]:
            max_char = ch
    return max_char

# INPUT
print(mostFrequentChar("success"))  # s

File: week_1/day-5-string-hashing/variations_que/mostFrequentCharacter.js
function mostFrequentChar(s) {
    const freq = {};
    let maxChar = s[0];

    for (let ch of s) {
        freq[ch] = (freq[ch] || 0) + 1;
        if (freq[ch] > freq[maxChar]) maxChar = ch;
    }
    return maxChar;
}

console.log(mostFrequentChar("success")); // s
